### PR TITLE
log history. requires information from backend

### PR DESCRIPTION
**Pull request by tcd3**

### CODE REVIEW

File: src/components/custom/PassedBadge.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export function PassedBadge() {
+	return (
+		<div className="justify-center items-center h-fit px-2 py-0.5 rounded-lg text-sm bg-green-300 text-green-600">
+			Published
```
Comment (ben-harper27) [inline src/components/custom/PassedBadge.js:4]: Shouldn't this be `Approved`?

File: src/components/custom/RejectedBadge.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+export function RejectedBadge() {
+	return (
+		<div className="justify-center items-center h-fit px-2 py-0.5 rounded-lg text-sm bg-red-300 text-red-600">
+			Published
```
Comment (ben-harper27) [inline src/components/custom/RejectedBadge.js:4]: This shouldn't be `Published` but should be `Rejected`. Also, the styling of this is almost the same as the previous - we shouldn't be making a new component for this. We also need badges for `Contributed` and `Not contributed` looking at the figma.

File: src/components/custom/VoteCard.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
+import {Card} from "@/components/ui/card";
+import Image from "next/image";
+import {RejectedBadge} from "@/components/custom/RejectedBadge";
+import {PassedBadge} from "@/components/custom/PassedBadge";
+
+export function VoteCard({change, changeVotes, user}) {
```
Comment (ben-harper27) [inline src/components/custom/VoteCard.js:6]: This should probably be called `ChangeCard` as it represents a change

File: src/pages/projects/[pid]/document/[documentId]/vote/index.js [hunk 4]
```diff
@@ -1,93 +1,123 @@
...
-      document: document,
-      changesWithVotes: changesWithVotes,
-    },
-  };
+	return {
+		props: {
+			project: project,
+			document: document,
+			changesWithVotes: changesWithVotes,
+			userProfile: userProfile,
```
Comment (ben-harper27) [inline src/pages/projects/[pid]/document/[documentId]/vote/index.js:120]: We have this in the store and so shouldn't be passing it down the stack.

File: src/pages/projects/[pid]/document/[documentId]/vote/index.js [hunk 5]
```diff
@@ -1,93 +1,123 @@
...
+				<section className="mt-3 max-md:max-w-full">
+					<div className="flex gap-5 max-md:flex-col max-md:gap-0">
+						{changesWithVotes.map((change, index) => (
+							((change.closed || change.merged) ? (
+								<VoteCard
+									key={index}
+									className="py-8 border-b-2 cursor-pointer"
+									change={change}
+									changeVotes={change.votes}
+									user={userProfile}
```
Comment (ben-harper27) [inline src/pages/projects/[pid]/document/[documentId]/vote/index.js:55]: No need to pass userProfile, just get it out of the store in a ChangeCard

File: src/pages/projects/[pid]/document/[documentId]/vote/index.js [hunk 6]
```diff
@@ -1,93 +1,123 @@
...
 import {getDocument, getDocumentChanges} from "@/lib/document";
 import {getChangeVotes} from "@/lib/change";
+import {getUserProfile} from "@/lib/user";
+import {NewLayout} from "@/components/NewLayout";
+import {VoteCard} from "@/components/custom/VoteCard";
 
-export default function Index({ project, document, changesWithVotes}) {
-  const router = useRouter();
+export default function Index({project, document, changesWithVotes, userProfile}) {
+	const router = useRouter();
```
Comment (ben-harper27) [inline src/pages/projects/[pid]/document/[documentId]/vote/index.js:11]: Undo adding the prop

**Review by ben-harper27 — CHANGES_REQUESTED** [review]

File: src/components/custom/VoteCard.js (hunk 3 above)
Comment (tcd3) [reply src/components/custom/VoteCard.js:6]: fixed

File: src/components/custom/PassedBadge.js (hunk 1 above)
Comment (tcd3) [reply src/components/custom/PassedBadge.js:4]: Yes. Fixed

File: src/components/custom/RejectedBadge.js (hunk 2 above)
Comment (tcd3) [reply src/components/custom/RejectedBadge.js:4]: Resolved

**Pull request merged**